backend/src/modules/task/controllers/task.controller.ts

```diff
@@ -40,8 +42,36 @@ export class TaskController {
   }
 
   @Get()
-  async getAll() {
-    return await this.taskService.getAllTasks();
+  async getAllTasks(
+    @Query('status') status?: string, 
+    @Query('createdBy') createdBy?: string,
+    @Query('assignedTo') assignedTo?: string,
+  ) {
+    const where: any = { deletedAt: null };
+
+    if (
+      status &&
+      status !== 'ALL' &&
+      Object.values(Status).includes(status as Status)
+    ) {
+      where.status = status as Status;
+    }
+
+    if (createdBy) {
+      const createdByInt = parseInt(createdBy, 10);
+      if (!isNaN(createdByInt)) where.createdBy = createdByInt;
+    }
+    if (assignedTo) {
+      const assignedToInt = parseInt(assignedTo, 10);
+      if (!isNaN(assignedToInt)) where.assignedTo = assignedToInt;
+    }
+
```

Comment:
please move all your logic inside service, we dont keep business or query logics inside controller